.github/workflows/sam-automation-tests.yaml

```diff
@@ -0,0 +1,118 @@
+name: sam-automation-tests
+
+on:
+  pull_request:
+    paths-ignore: ['**.md']
+  push:
+    paths-ignore: ['**.md']
+
+jobs:
+  sam-build-tag-publish-job:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    outputs:
+      custom-version-json: ${{ steps.render-data.outputs.custom-version-json }}
+    steps:
+      - name: Print branch
+        run: |
+          push_branch=${{ github.ref }}
+          pr_branch=refs/heads/${{ github.head_ref }}
+          branch=$push_branch || $pr_branch
+          echo $branch
+
+      - name: Extract branch from push event
+        if: github.event_name == 'push'
+        id: extract-branch-push
+        env:
+          GITHUB_CONTEXT: ${{ toJSON(github) }}
+        run: |
+          echo ::set-output name=ref::${{ github.ref }}
+
+      - name: Extract branch from PR event
+        if: github.event_name == 'pull_request'
+        id: extract-branch-pr
+        env:
+          GITHUB_CONTEXT: ${{ toJSON(github) }}
+        run: |
+          echo ::set-output name=ref::refs/heads/${{ github.head_ref }}
+
+      - name: dispatch to terra-github-workflows
```

Comment:
recomment `dispatch build to terra-github-workflows` to be more description from the high level.

Reply:
Updated comment.